### ISSUE TITLE
Basic structure

### ISSUE THREAD

**Issue by ajakubo1** [EnviroMonitor/EnviroMonitorFrontend#2]
@zen 

Can you let me know what should the frontend consist of? I can see there:

- Login Screen
- List/Add project
- List/Add station (though I can incorporate that with project listing, I think)
- Data visualisation (some charts and tables)

I know that there are no mockups, so I won't focus on css-es right now, I'll just do the html structure + event handling.

**Comment by zen** [comment]
It's an educational project, so information is most important. We will need to feed system with some data (some data generator would be nice), and first concentrate on following aspects:
- choosing data to be presented (configurable)
- presentation of the data
- information about the "project" (would be nice to integrate that with some simple django CMS)
- project news (simple blog, also preferably based on django

I'm not sure if we need admin backend other than django one, it should be enough for station and project owners to manage it there. We might add some nice skin with JS bells and whistles to django admin. 

I will try to create some mockups for main project screen with data visualisation. For start we would need pm2.5 and pm10 from selected area, with average based on visible timeline. As for timelines, I would suggest last hour, 12h, day, week, month. Or, if possible, some dynamic slider for changing dates of graph.

**Comment by ajakubo1** [comment]
Ah, Ok, so it's not a strict application which we want here. We have two parts: the blog/news page and the application itself.

So - the application part - objects in project/station should be managed by django-admin.

From my point of view, we can use React as frontend, but we only focus on data presentation there. And we use admin for configuration.

I'll try to checkout django CMS (but to tell you the truth I didn't work with it at all). I'm also not totally sure if we should merge blog/information part with application itself? Those are totally separate things for me. If someone want's to use the detectors with the web page that we are doing - he will get the blogging platform attached to it? I don't think we should merge those together.

And with that approach we can use any cms for content (like wordpress, drupal, or maybe some static-generation cms)? Unless you really want it in django (I still think that those should be two totally separate things).

**Comment by zen** [comment]
Yes, good points, we should have kind of main page, where all data is presented, according to current focus on map. There is some concept here:
http://looko2.com/heatmap.php

What I would add is bar at the bottom displaying averages for current map focus. The more you zoom, the more local data you get. What do you think about this?

For blog etc. we will probably need to attach it to project, and then show links on detail view for stations associated with projects. I think more detailed requirements would come once we have some users.

**Comment by zen** [comment]
Also, detailed view:
http://looko2.com/tracker.php

I'm thinking how to present current level using color hints, like green - good, orange - moderate, - red - bad, claret - critical

**Comment by ajakubo1** [comment]
Yeah - this makes a lot of sense to me! Clean and simple.

Also - we could make pins clickable to display current information per location with a chart or something? You know, so users wouldn't have to zoom all the way.

**Comment by lechup** [comment]
What do You think about https://airly.eu/map/ ? Of course we can skip "animate last 24h" ...

**Comment by zen** [comment]
Sure, this looks very good, but be very careful with "inspiration".

**Comment by lechup** [comment]
I guess this one includes #3?
